### ISSUE TITLE
slice flag value don't append to default values from ENV or file

### ISSUE THREAD

**Issue by vipally** [urfave/cli#1239]
## my urfave/cli version is

v2.3.0

## Checklist

* [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
* [x] Did you perform a search about this problem? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## Dependency Management

- My project is using go modules.
- My project is automatically downloading the latest version.

## Describe the bug

Slice flag value don't append to default values from ENV or file.

## To reproduce

I have put the test code [here](https://github.com/vipally/glab/blob/master/lab26/lab26.go#L38)
```go
package main

import (
	"fmt"
	"os"

	"github.com/urfave/cli/v2"
)

func main() {
	os.Args = []string{"multi_values",
		"--stringSclice", "parsed1", "--stringSclice", "parsed3",
		"--float64Sclice", "13.3", "--float64Sclice", "15.5",
		"--int64Sclice", "13", "--int64Sclice", "15",
		"--intSclice", "13", "--intSclice", "15",
	}
	os.Setenv("ENV_CLI_STRING_SLICE", "foo,bar")
	os.Setenv("ENV_CLI_FLOAT64_SLICE", "23.3,24.4")
	os.Setenv("ENV_CLI_INT64_SLICE", "23,24")
	os.Setenv("ENV_CLI_INT_SLICE", "23,24")
	app := cli.NewApp()
	app.Name = "multi_values"
	app.Flags = []cli.Flag{
		&cli.StringSliceFlag{Name: "stringSclice", EnvVars: []string{"ENV_CLI_STRING_SLICE"}},
		&cli.Float64SliceFlag{Name: "float64Sclice", EnvVars: []string{"ENV_CLI_FLOAT64_SLICE"}},
		&cli.Int64SliceFlag{Name: "int64Sclice", EnvVars: []string{"ENV_CLI_INT64_SLICE"}},
		&cli.IntSliceFlag{Name: "intSclice", EnvVars: []string{"ENV_CLI_INT_SLICE"}},
	}
	app.Action = func(ctx *cli.Context) error {
		for i, v := range ctx.FlagNames() {
			fmt.Printf("%d-%s %#v\n", i, v, ctx.Value(v))
		}
		return ctx.Err()
	}

	_ = app.Run(os.Args)

	// bug: slice flag value don't append to default values from ENV or file
	/*
		0-float64Sclice cli.Float64Slice{slice:[]float64{23.3, 24.4, 13.3, 15.5}, hasBeenSet:true}	// bug: expect []float64{13.3, 15.5}
		1-int64Sclice cli.Int64Slice{slice:[]int64{23, 24, 13, 15}, hasBeenSet:true}				// bug: expect []int64{13, 15}
		2-intSclice cli.IntSlice{slice:[]int{23, 24, 13, 15}, hasBeenSet:true}						// bug: expect []int{13, 15}
		3-stringSclice cli.StringSlice{slice:[]string{"parsed1", "parsed3"}, hasBeenSet:true}
	*/
}

```

## Observed behavior

```
0-float64Sclice cli.Float64Slice{slice:[]float64{23.3, 24.4, 13.3, 15.5}, hasBeenSet:true}
1-int64Sclice cli.Int64Slice{slice:[]int64{23, 24, 13, 15}, hasBeenSet:true}
2-intSclice cli.IntSlice{slice:[]int{23, 24, 13, 15}, hasBeenSet:true}
3-stringSclice cli.StringSlice{slice:[]string{"parsed1", "parsed3"}, hasBeenSet:true}
```

## Expected behavior

```
0-float64Sclice cli.Float64Slice{slice:[]float64{13.3, 15.5}, hasBeenSet:true}
1-int64Sclice cli.Int64Slice{slice:[]int64{13, 15}, hasBeenSet:true}
2-intSclice cli.IntSlice{slice:[]int{13, 15}, hasBeenSet:true}
3-stringSclice cli.StringSlice{slice:[]string{"parsed1", "parsed3"}, hasBeenSet:true}
```

## Additional context

I have put the test code [here](https://github.com/vipally/glab/blob/master/lab26/lab26.go#L38)

## Want to fix this yourself?

We'd love to have more contributors on this project! If the fix for this bug is easily explained and very small, free free to create a pull request for it.
## Run `go version` and paste its output here

```
go version go1.15.6 windows/amd64
```

## Run `go env` and paste its output here

```sh
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\ally\AppData\Local\go-build
set GOENV=C:\Users\ally\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GOMODCACHE=C:\Users\ally\go\pkg\mod
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=C:\Users\ally\go
set GOPRIVATE=
set GOPROXY=https://goproxy.cn
set GOROOT=c:\go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=c:\go\pkg\tool\windows_amd64
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=C:\Users\ally\AppData\Local\Temp\go-build790788990=/tmp/go-build -gno-record-gcc-switches
```

**Comment by lynncyrin** [comment]
I'm slightly worried that this might be a breaking change for someone somewhere, but at a basic level the expected behavior you describe is definitely correct 👍🏼

**Comment by meatballhat** [comment]
Actually closed via #1240
